src/GUI/Preset.hpp

```diff
@@ -21,23 +32,28 @@ class Preset;
 
 using Presets = std::vector<Preset>;
 
+class PresetEditor;
+
 class Preset {
 public:
+    friend class PresetEditor;
```

Comment:
Rather than giving PresetEditor friend access, we can instead make a method config() that returns a config_ptr that is a transformed weak_ptr.

Reply:
Better answer: new config_ref which is a std::weak_ptr<Slic3r::Config>. 

Usage in client functions that were asking for a dirty config is:
````
if (config_ptr x = preset.config()) { 
    ...
}
````